wagtail/test/utils/page_tests.py

```diff
@@ -148,3 +162,288 @@ def assertAllowedParentPageTypes(self, child_model, parent_models, msg=None):
+        user: Optional[AbstractBaseUser] = None,
+        msg: Optional[str] = None,
+    ):
+        """
+        Asserts that the page preview view can be loaded for ``page`` without raising a fatal error.
+
+        For page types that support multiple preview modes, ``mode`` can be used to specify the preview mode to be tested.
+
+        When ``user`` is provided, the test is conducted with them as the active user. Otherwise, a superuser is created and used for the test.
+
+        To load the preview, the test client needs to make a ``POST`` request including all required field data in the request body.
+        If ``post_data`` is provided, that will be used for this purpose. If not, the method will attempt to extract this data from the page edit view.
+        """
+        if not user:
+            if not hasattr(self, "_pageispreviewable_superuser"):
+                self._pageispreviewable_superuser = self.create_superuser(
+                    "assertpageispreviewable"
+                )
+            user = self._pageispreviewable_superuser
+
+        self.client.force_login(user, AUTH_BACKEND)
+
+        if post_data is None:
+            edit_path = reverse("wagtailadmin_pages:edit", kwargs={"page_id": page.id})
+            html = self.client.get(edit_path).content.decode()
+            post_data = querydict_from_html(html, form_id="page-edit-form")
+
+        preview_path = reverse(
+            "wagtailadmin_pages:preview_on_edit", kwargs={"page_id": page.id}
+        )
+        try:
+            response = self.client.post(
+                preview_path, data=post_data, QUERYSTRING=f"mode={mode}"
+            )
+            self.assertEqual(response.status_code, 200)
+            self.assertJSONEqual(
+                response.content.decode(),
+                {"is_valid": True, "is_available": True},
+            )
+        except Exception as e:
+            self.client.logout()
+            msg = self._formatMessage(
+                msg,
+                'Failed to load preview for %(page_type)s "%(page)s" with mode="%(mode)s":\n%(exc)s'
+                % {
+                    "page_type": type(page).__name__,
+                    "page": page,
+                    "mode": mode,
+                    "exc": e,
+                },
+            )
+            raise self.failureException(msg)
+
+        try:
+            self.client.get(preview_path, data={"mode": mode})
+        except Exception as e:
+            msg = self._formatMessage(
+                msg,
+                'Failed to load preview for %(page_type)s "%(page)s" with mode="%(mode)s":\n%(exc)s'
+                % {
+                    "page_type": type(page).__name__,
+                    "page": page,
+                    "mode": mode,
+                    "exc": e,
+                },
+            )
+            raise self.failureException(msg)
+        finally:
+            self.client.logout()
+
+
+class WagtailPageTests(WagtailPageTestCase):
```

Comment:
Note: the docs now promote the use of `WagtailPageTestCase` ([ref](https://wagtail--9294.org.readthedocs.build/en/9294/advanced_topics/testing.html)). 

Do we want to deprecate this? given it creates and logs a user in, in addition to all the assertions?

Reply:
I have added a note about this & pushed.

Reply:
I think it's fine that we have the two test cases, as I'm sure there are times when having the user created and logged in for each test is useful. It's just wasteful if the intention is to only use these assertions, as that user will be logged out :)